library/ecp.c

```diff
@@ -3356,6 +3356,30 @@ int mbedtls_ecp_check_pub_priv(
     return( ret );
 }
 
+/*
+ * Export generic key-pair parameters.
+ */
+int mbedtls_ecp_export(const mbedtls_ecp_keypair *key, mbedtls_ecp_group *grp,
+                       mbedtls_mpi *d, mbedtls_ecp_point *Q)
+{
+    int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
+    ECP_VALIDATE_RET( key != NULL );
```

Comment:
Minor: `VALIDATE` macros do nothing since 3.0. We haven't gone the trouble of removing them yet, but we shouldn't add them in new function. The plan is to remove `NULL` checks and keep only checks like "this `int` argument should be 0 or 1" or similar, but with plain `if( ... ) return MBEDTLS_ERR_XXX_BAD_INPUT_DATA;`, no validation macros.

Reply:
I was wondering if this is needed as it was used in other functions, but does nothing. Added it for consistency assuming that maybe the intention is that user can redefine macro to enable null checks. 
Removed.
Do we have an issue for this? Maybe I could remove these checks?

Reply:
> assuming that maybe the intention is that user can redefine macro to enable null checks.

Yes, that was the case in 2.x - but now in 3.0 it's unconditionally dead code.

We do have an issue: https://github.com/ARMmbed/mbedtls/issues/4681